### PR TITLE
WF-65182-Apply base style to table style UG documentation

### PR DESCRIPTION
**Pull request by sharonaumao**
<table>
<tr>
<td>
Task Link
</td>
<td>
<a href="https://syncfusion.atlassian.net/browse/WF-65182">https://syncfusion.atlassian.net/browse/WF-65182</a>
</td>
</tr>
<tr>
<td>
Description
</td>
<td>
Add UG document for apply base style to table styles using BuilinTableStyle
</td>
</tr>
<td>
Content review confirmation
</td>
<td>
 Not yet
</td>
</tr>
</table>

### CODE REVIEW

File: File-Formats/DocIO/Working-with-Tables.md [hunk 1]
```diff
@@ -2247,6 +2247,370 @@ document.Close();
...
+Table styles can be based on other table styles also. When applying base style, the new style will inherit the values of the base style that are not explicitly redefined in the new style. You can apply custom table style or built-in table as a base for the table style using BuiltinTableStyle API.
+
+The following code example illustrates how to apply built-in and custom table styles as base style for another custom table.
+
+{% tabs %}
+
+{% highlight c# tabtitle="C#" %}
+//Create a new Word document.
+using (WordDocument document = new WordDocument())
+{
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Tables.md:2261]: //Add one section and paragraph in document.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 2]
```diff
@@ -2247,6 +2247,370 @@ document.Close();
...
+    //Save the stream as a file in the device and invoke it for viewing.
+    Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Sample.docx", "application/msword", outputStream);
+    //Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
+    //https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+}
+{% endhighlight %}
+
+{% endtabs %}
+
+You can download a complete working sample from [GitHub]().
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Working-with-Tables.md:2612]: We will add this line later, once sample merged in live. At now, remove this line

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/Working-with-Tables.md (hunk 2 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: @MohanaselvamJothi -I have removed it

File: File-Formats/DocIO/Working-with-Tables.md (hunk 1 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md:2261]: @MohanaselvamJothi -I have added it

File: File-Formats/DocIO/Working-with-Tables.md [hunk 3]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    tableStyle2.ApplyBaseStyle("CustomStyle2");
+    //Apply the custom table style to the table.
+    table.ApplyStyle("CustomStyle3");
+
+    //Save the Word document to the MemoryStream.
+    MemoryStream outputStream = new MemoryStream();
+    document.Save(outputStream, FormatType.Docx);
+    //Save the stream as a file in the device and invoke it for viewing.
+    Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Sample.docx", "application/msword", outputStream);
+    //Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2610]: //Please download the helper files from the following link to save the stream as a file and open the file for viewing in the Xamarin platform.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 4]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    firstRowStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle.ParagraphFormat.HorizontalAlignment = HorizontalAlignment.Center;
+    //Apply conditional formatting for odd row.
+    oddRowBandingStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.OddRowBanding);
+    oddRowBandingStyle.CharacterFormat.TextColor = Color.Red;
+
+    //Add a new custom table style.
+    WTableStyle tableStyle2 = document.AddTableStyle("CustomStyle3") as WTableStyle;
+    tableStyle2.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2448]: //Apply conditional formatting for the first row.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 5]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    table[0, 0].AddParagraph().AppendText("Row 1 Cell 1");
+    table[0, 1].AddParagraph().AppendText("Row 1 Cell 2");
+    table[1, 0].AddParagraph().AppendText("Row 2 Cell 1");
+    table[1, 1].AddParagraph().AppendText("Row 2 Cell 2");
+    table[2, 0].AddParagraph().AppendText("Row 3 Cell 1");
+    table[2, 1].AddParagraph().AppendText("Row 3 Cell2");
+    //Add a new custom table style.
+    WTableStyle tableStyle = document.AddTableStyle("CustomStyle1") as WTableStyle;
+    tableStyle.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2276]: //Apply conditional formatting for the first row.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 6]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    ConditionalFormattingStyle firstRowStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle.CharacterFormat.Bold = true;
+    //Apply conditional formatting for odd row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2279]: //Apply conditional formatting for the odd row.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 7]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    ConditionalFormattingStyle oddRowBandingStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.OddRowBanding);
+    oddRowBandingStyle.CharacterFormat.Italic = true;
+    // Apply built in table style as base style for CustomStyle1.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2282]: // Apply built-in table style as base style for CustomStyle1.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 8]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    tableStyle.ApplyBaseStyle(BuiltinTableStyle.TableContemporary);
+    //Applies the custom table style to the table
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2284]: //Apply the custom table style to the table.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 9]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    table = document.LastSection.AddTable() as WTable;
+    table.ResetCells(3, 2);
+    table[0, 0].AddParagraph().AppendText("Row 1 Cell 1");
+    table[0, 1].AddParagraph().AppendText("Row 1 Cell 2");
+    table[1, 0].AddParagraph().AppendText("Row 2 Cell 1");
+    table[1, 1].AddParagraph().AppendText("Row 2 Cell 2");
+    table[2, 0].AddParagraph().AppendText("Row 3 Cell 1");
+    table[2, 1].AddParagraph().AppendText("Row 3 Cell2");
+
+    //Adds a new custom table style.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2298]: //Adds a new custom table style.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 10]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    tableStyle = document.AddTableStyle("CustomStyle2") as WTableStyle;
+    tableStyle.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2301]: //Apply conditional formatting for the first row.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 11]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    firstRowStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle.ParagraphFormat.HorizontalAlignment = HorizontalAlignment.Center;
+    //Apply conditional formatting for odd row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2304]: //Apply conditional formatting for the odd row.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 12]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    oddRowBandingStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.OddRowBanding);
+    oddRowBandingStyle.CharacterFormat.TextColor = Color.Red;
+
+    //Add a new custom table style.
+    WTableStyle tableStyle2 = document.AddTableStyle("CustomStyle3") as WTableStyle;
+    tableStyle2.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2311]: //Apply conditional formatting for the first row.

File: File-Formats/DocIO/Working-with-Tables.md [hunk 13]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    ConditionalFormattingStyle firstRowStyle2 = tableStyle2.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle2.CellProperties.BackColor = Color.Blue;
+    //Apply conditional formatting for odd row.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Working-with-Tables.md:2314]: //Apply conditional formatting for the odd row.

**Review by florenceliliana — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/Working-with-Tables.md [hunk 14]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
 
 You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/DocIO-Examples/tree/main/Tables/Apply-custom-table-style).
 
+### Apply Base Style
+
+Table styles can be based on other table styles also. When applying base style, the new style will inherit the values of the base style that are not explicitly redefined in the new style. You can apply custom table style or built-in table as a base for the table style using BuiltinTableStyle API.
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: @florenceliliana -Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 15]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+
+The following code example illustrates how to apply built-in and custom table styles as base style for another custom table.
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: @florenceliliana -Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 16]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    tableStyle.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
+    ConditionalFormattingStyle firstRowStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle.CharacterFormat.Bold = true;
+    //Apply conditional formatting for odd row.
+    ConditionalFormattingStyle oddRowBandingStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.OddRowBanding);
+    oddRowBandingStyle.CharacterFormat.Italic = true;
+    // Apply built in table style as base style for CustomStyle1.
+    tableStyle.ApplyBaseStyle(BuiltinTableStyle.TableContemporary);
+    //Applies the custom table style to the table
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 17]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    //Apply custom table style as base style for another custom table style.
+    tableStyle2.ApplyBaseStyle("CustomStyle2");
+    //Apply the custom table style to the table.
+    table.ApplyStyle("CustomStyle3");
+
+    //Save the Word document to the MemoryStream.
+    await document.SaveAsync(stream, FormatType.Docx);
+    //Save the stream as Word file in local machine.
+    Save(stream, "Sample.docx");
+    //Please refer the below link to save Word document in UWP platform
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 18]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    //https://help.syncfusion.com/file-formats/docio/create-word-document-in-uwp#save-word-document-in-uwp
+}
+{% endhighlight %}
+
+{% highlight c# tabtitle="ASP.NET Core" %}
+//Create a new Word document.
+using (WordDocument document = new WordDocument())
+{
+    //Add one section and paragraph in document.
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 9 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 10 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 11 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 12 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 13 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 19]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    tableStyle.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
+    ConditionalFormattingStyle firstRowStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle.CharacterFormat.Bold = true;
+    //Apply conditional formatting for odd row.
+    ConditionalFormattingStyle oddRowBandingStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.OddRowBanding);
+    oddRowBandingStyle.CharacterFormat.Italic = true;
+    // Apply built in table style as base style for CustomStyle1.
+    tableStyle.ApplyBaseStyle(BuiltinTableStyle.TableContemporary);
+    //Applies the custom table style to the table
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 20]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    //Apply custom table style as base style for another custom table style.
+    tableStyle2.ApplyBaseStyle("CustomStyle2");
+    //Apply the custom table style to the table.
+    table.ApplyStyle("CustomStyle3");
+
+    //Save the Word document to the MemoryStream.
+    MemoryStream outputStream = new MemoryStream();
+    document.Save(outputStream, FormatType.Docx);
+    outputStream.Position = 0;
+    //Download Word document in the browser.
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md [hunk 21]
```diff
@@ -2247,6 +2247,373 @@ document.Close();
...
+    tableStyle.TableProperties.RowStripe = 1;
+    //Apply conditional formatting for first row.
+    ConditionalFormattingStyle firstRowStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.FirstRow);
+    firstRowStyle.CharacterFormat.Bold = true;
+    //Apply conditional formatting for odd row.
+    ConditionalFormattingStyle oddRowBandingStyle = tableStyle.ConditionalFormattingStyles.Add(ConditionalFormattingType.OddRowBanding);
+    oddRowBandingStyle.CharacterFormat.Italic = true;
+    // Apply built in table style as base style for CustomStyle1.
+    tableStyle.ApplyBaseStyle(BuiltinTableStyle.TableContemporary);
+    //Applies the custom table style to the table
```
[the remark this answers is not in the document]
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 3 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 4 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 5 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 6 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 7 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

File: File-Formats/DocIO/Working-with-Tables.md (hunk 8 above)
Comment (sharonaumao) [reply File-Formats/DocIO/Working-with-Tables.md]: Addressed

**Review by MohanaselvamJothi — APPROVED** [review]

**Pull request merged**